gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -150,6 +150,11 @@ def clone
   sig {returns(SerializedVal).checked(:never)}
   def serialize
     assert_bound!
+    _serialized_val
+  end
+
+  sig {returns(SerializedVal).checked(:never)}
+  def _serialized_val
```

Comment:
There's a reason why you can only call `serialize` if `_bind_name` has been called: not all enums have a serialized name when they're initialized (i.e., `A = new` is valid and has no serialized_val).

I'd **rather not punch a hole here** and expose the possibility for someone to misuse this.

The error messages are going to be equally bad: either we print something like `[#<MyEnum:0x00007f8b3d81e898>]` or we print `[nil]`. Arguably `MyEnum` is better because at least it shows the name of the enum.

With that in mind, can you add some tests for that code path that use `A = new` instead of `A = new('a')`?

Reply:
Ack. If I print out `orphaned_instances` directly [below](https://github.com/sorbet/sorbet/pull/2605/files#diff-64470ec674da0d4beda81e2e5313b6ccR351), it prints out,
```
"Enum values must be assigned to constants: [#<::__UNINITIALIZED__>]"
```
which is hard to read. This hole was meant to solve that. Can you suggest a better way? 
Thanks

Reply:
One option would be
```ruby
raise "Enum values must be assigned to constants: #{orphaned_instances.map {|v| v.instance_variable_get('@serialized_val')}}"
```
Would that be better than punching this hole with a method?
Alternately, `_serialized_val` could be private and it could `send(:serialized_val)`.

Or maybe there's a better way that I'm not seeing.

Thanks

Reply:
(I went with `instance_variable_get` for now.)